File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> counts;
        vector<int> res;
        for(int num : nums){
            counts[num] += 1;
        }

        vector<pair<int,int>> arr;
        for(const auto& p : counts) {
            arr.push_back({p.second, p.first});
        }

        sort(arr.begin(), arr.end());

        for (int i = arr.size() - 1; k > 0; --k) {
            res.push_back(arr[i].second);
            --i;
        }

        return res;

    }
};

File: Data Structures & Algorithms/sliding-window-maximum/submission-4.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        q = deque()
        res = []
        i = j = 0

        while j < len(nums):
            while q and nums[j] > q[-1][0]:
                q.pop()
            q.append((nums[j], j))

            if i > q[0][1]:
                q.popleft()

            if j >= k - 1:
                res.append(q[0][0])
                i += 1

            j += 1

        return res
            